src/Downloader.php

```diff
@@ -16,27 +16,37 @@
 use Composer\DependencyResolver\Operation\OperationInterface;
 use Composer\DependencyResolver\Operation\UninstallOperation;
 use Composer\DependencyResolver\Operation\UpdateOperation;
-use Composer\Downloader\TransportException;
 use Composer\IO\IOInterface;
 use Composer\Json\JsonFile;
+use Composer\Util\Http\Response as ComposerResponse;
 use Composer\Util\HttpDownloader;
 use Composer\Util\Loop;
 
 /**
  * @author Fabien Potencier <fabien@symfony.com>
+ * @author Nicolas Grekas <p@tchwork.com>
  */
 class Downloader
 {
-    private static $DEFAULT_ENDPOINT = 'https://flex.symfony.com';
-    private static $MAX_LENGTH = 1000;
+    private const DEFAULT_ENDPOINTS = [
+        'https://api.github.com/repos/symfony/recipes/contents/index.json?ref=flex/main' => [],
```

Comment:
@nicolas-grekas would it make sense for backwards compatibility to rename the current `main` and `master` branches to `1.x` so if there is in future a `2.x` release which maybe use another `.json` format which will not break?

Reply:
We're talking about the branch name of recipes. There will be no v2 as there is no v1 either :)

Reply:
What @alexander-schranz means I think is: what if we refactor the JSON api in the future? (e.g. another json format to support cool new things)

If we don't version the URL, it means we force all apps to directly upgrade to the latest Flex version or to make all JSON format changes backwards compatible.

Reply:
the version would then belong to what's inside the json, not to the URL
I think all is future proof already:
- different format of recipes (eg php-dsl) = new endpoint, eg flex/main replaced by flex/main-php
- new API payloads = new keys in index.json, possibly with a version, but only the day we actually need it imho

Reply:
If you are sure there will be no issue with ref to main/master then its good :)